libopae/pluginmgr.c

```diff
@@ -190,15 +240,166 @@ int opae_plugin_mgr_finalize_all(void)
 	}
 
 	initialized = 0;
-
+	opae_plugin_mgr_reset_cfg();
 	opae_mutex_unlock(res, &adapter_list_lock);
 
 	return errors;
 }
 
-STATIC int opae_plugin_mgr_parse_config(/* json_object *jobj */)
+#define JSON_GET(_jobj, _key, _jvar)                                           \
+	do {                                                                   \
+		if (!json_object_object_get_ex(_jobj, _key, _jvar)) {          \
+			OPAE_ERR("Error getting object: %s", _key);            \
+			return 1;                                              \
+		}                                                              \
+	} while (0)
+
+#define MAX_PLUGIN_CFG_SIZE 1024
+STATIC int process_plugin(const char *name, json_object *j_config)
+{
+	plugin_cfg *cfg = &plugin_list[plugin_count];
+	const char *stringified = NULL;
+	json_object *j_module = NULL;
+	json_object *j_plugin_cfg = NULL;
+	json_object *j_enabled = NULL;
+	JSON_GET(j_config, "module", &j_module);
```

Comment:
I think you mean module as in module .so file.  I'm concerned that some users won't understand this convention.  Could we use plugin for clarity?

Reply:
OK. Sounds good. One of the reasons for doing this in a feature branch is to hash out the things like this.